pyproject.toml

```diff
@@ -46,6 +46,7 @@ select = [
     "N", # pep8-naming
     "PGH", # pygrep-hooks
     "RUF", # Ruff-specific and unused-noqa
+    "SIM", # flake8-simplify
```

Comment:
I still think I'd rather explicitly list the rules we think are useful here, rather than enabling the whole category and disabling specific rules. This category is just too hit-and-miss for me: some of the rules are pretty clear wins, but quite a few make some very opinionated suggestions that have high false-positive rates. Even if some of those rules don't have any false positives on typeshed today, there's a chance we might add code in the future that they'd emit false positives on, or there's a chance that Ruff might stabilise new `SIM` rules that have false positives on typeshed code.

If lint rules end up irritating us (or confusing contributors) more than they actually help us write better code, then it defeats the purpose of a linter, in my opinion :-)

Reply:
Updated to be more specific about rules selection
I listed the rules I didn't select in the PR's description